packages/endo/src/archive.js

```diff
@@ -0,0 +1,158 @@
+/* global StaticModuleRecord */
+/* eslint no-shadow: 0 */
+
+import { writeZip } from "./zip.js";
+import { resolve, join } from "./node-module-specifier.js";
+import { compartmentMapForNodeModules } from "./compartmap.js";
+import { search } from "./search.js";
+import { assemble } from "./assemble.js";
+
+const { entries, fromEntries } = Object;
+
+const q = JSON.stringify;
+
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+const makeRecordingImportHookMaker = (
+  read,
+  baseLocation,
+  manifest,
+  errors
+) => packageLocation => {
```

Comment:
huh, I find this multi-line form of the nested-fat-arrow syntax easier to mentally parse as a function returning a function, for some reason. Maybe I can pretend that line 22 starts with `function (packageLocation)` more readily when there aren't a bunch of arguments for some other function on the same line.

Reply:
Posting a follow-up to once again shy ML away.